src/main/java/hudson/maven/reporters/TestMojo.java

```diff
@@ -167,21 +168,34 @@ public boolean canRunTests(MojoInfo mojo) {
     }
     
     @CheckForNull public Iterable<File> getReportFiles(MavenProject pom, MojoInfo mojo) throws ComponentConfigurationException {
+        File reportsDir = getReportsDirectory(pom, mojo);
+        if (reportsDir != null && reportsDir.exists())
+            return getReportFiles(reportsDir, getFileSet(reportsDir));
+        
+        return null;
+    }
+
+    /**
+     * Returns the location of test reports created by the specified MOJO.
+     * @param pom The project model.
+     * @param mojo The MOJO.
+     * @return The directory containing the test reports.
+     * @throws ComponentConfigurationException if unable to retrieve the report directory from the MOJO configuration.
+     */
+    protected File getReportsDirectory(MavenProject pom, MojoInfo mojo) throws ComponentConfigurationException {
+    	// [JENKINS-31258] Allow unknown MOJOs to contribute test results in arbitrary locations by setting a Maven property.
+		String reportsDirectoryOverride = getReportsDirectoryOverride(mojo);
+		if (reportsDirectoryOverride != null)
```

Comment:
This file has used spaces throughout since its inception by Christoph Kutzinski on 06/01/2013. The Jenkins 'code of conduct' requests that one mainain stylistic consistency and explicitly bans auto-reformatting.